internal/tofu/node_resource_plan_instance.go

```diff
@@ -320,18 +320,22 @@ func (n *NodePlannableResourceInstance) managedResourceExecute(ctx context.Conte
 	} else {
 		if instanceRefreshState != nil {
 			prevCreateBeforeDestroy := instanceRefreshState.CreateBeforeDestroy
+			prevSkipDestroy := instanceRefreshState.SkipDestroy
 
 			// This change is usually written to the refreshState and then
 			// updated value used for further graph execution. However, with
 			// "refresh=false", refreshState is not being written and then
 			// some resources with updated configuration could be detached
 			// due to missaligned create_before_destroy in different graph nodes.
 			instanceRefreshState.CreateBeforeDestroy = n.Config.Managed.CreateBeforeDestroy || n.ForceCreateBeforeDestroy
-
-			if prevCreateBeforeDestroy != instanceRefreshState.CreateBeforeDestroy && n.skipRefresh {
-				diags = diags.Append(n.writeResourceInstanceState(ctx, evalCtx, instanceRefreshState, refreshState))
-				if diags.HasErrors() {
-					return diags
+			instanceRefreshState.SkipDestroy = !n.Config.Managed.Destroy
+
+			if n.skipRefresh {
+				if prevCreateBeforeDestroy != instanceRefreshState.CreateBeforeDestroy || prevSkipDestroy != instanceRefreshState.SkipDestroy {
```

Comment:
Could you provide more details in a comment here on what's the reason of this condition? For somebody that is not fully familiar with the skipDestroy, this part seems confusing.